modules/contact-form/grunion-contact-form.php

```diff
@@ -3554,6 +3619,12 @@ function render_field( $type, $id, $label, $value, $class, $placeholder, $requir
 			case 'date':
 				$field .= $this->render_date_field( $id, $label, $value, $field_class, $required, $required_field_text, $field_placeholder );
 				break;
+			case 'consent':
+				$consent_type    = 'explicit' === $this->get_attribute( 'consenttype' ) ? 'explicit' : 'implicit';
+				$consent_message = 'explicit' === $consent_type ? $this->get_attribute( 'explicitconsentmessage' ) : $this->get_attribute( 'implicitconsentmessage' );
```

Comment:
Maybe this could be moved inside the `render_consent_field` method, to keep all the logic in one place?

Reply:
clean it up